pkg/controller/cloudfront/distribution/lateinit.go

```diff
@@ -728,3 +737,48 @@ func lateInitLambdaFunctionAssociation(in *svcapitypes.LambdaFunctionAssociation
 	in.IncludeBody = awsclients.LateInitializeBoolPtr(in.IncludeBody, from.IncludeBody)
 	in.LambdaFunctionARN = awsclients.LateInitializeStringPtr(in.LambdaFunctionARN, from.LambdaFunctionARN)
 }
+
+func lateInitFunctionAssociations(in *svcapitypes.FunctionAssociations, from *svcsdk.FunctionAssociations) {
+	if len(from.Items) == 0 {
+		return
+	}
+
+	// If we have no function associations, late init the entire slice
+	if in.Items == nil {
+		in.Items = make([]*svcapitypes.FunctionAssociation, len(from.Items))
+		for i := range from.Items {
+			in.Items[i] = &svcapitypes.FunctionAssociation{}
+			lateInitFunctionAssociation(in.Items[i], from.Items[i])
+		}
+
+		return
+	}
+
+	// If we have some function associations, we need to late init each one of them
+	existing := make(map[string]*svcsdk.FunctionAssociation)
+	for _, o := range from.Items {
+		if o.FunctionARN == nil {
+			continue
+		}
```

Comment:
Please see the comment below regarding the chosen key. We would not expect the key type to be a pointer but although in the console UI, it's not a pointer type, the SDK expects a pointer type. We did not check what happens if you call the SDK function with a nil `EventType` but from what we observed from the console UI, this should not be allowed.
```suggestion
```

Reply:
Removing this nil check caused following error: 
```- lastTransitionTime: "2022-05-11T08:44:05Z"
    message: "update failed: cannot update Distribution in AWS: PreconditionFailed:
      The request failed because it didn't meet the preconditions in one or more request-header
      fields.\n\tstatus code: 412, request id: "
```
I added back with the `EventType`.